bookdown/08-interpretation.Rmd

```diff
@@ -1,10 +1,165 @@
+
+**Shapley**
+  
+```{r mod, message=FALSE, warning=FALSE, fig.cap='Plot of the results from Shapley. $\\phi$ gives the increase or decrease in probabilty given the values on the vertical axis',  fig.align='center'}
+x = penguins[which(names(penguins) != "species")]
+model = Predictor$new(learner, data = penguins, y = "species")
+x.interest <-data.frame(penguins[1,])
+shapley <- Shapley$new(model, x.interest = x.interest)
+plot(shapley)
+```
+
+The $\phi$ provides insight into the probability given the values on the vertical axis. For example, a penguin is less likely to be Gentoo if the bill_depth_=18.7 is and much more likely to be Adelie than Chinstrap.
+
+
+**FeatureImp**
+
+```{r imp, message=FALSE, warning=FALSE, fig.cap='Plot of the results from FeatureImp. FeatureImp visualizes the importance of features given the prediction model',  fig.align='center'}
+effect = FeatureImp$new(model, loss = "ce")
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+`FeatureImp` shows the level of importance of the features when classifying the penguins. It is clear to see that the `bill_length_mm` is of high importance and one should concentrate on different boundaries of this feature when attempting to classify the three species. 
+
+
+
+**Predicting**
+  
+It is also interesting to see how well a model has trained a data set. For this section, exactly as was recommended in section [2.4](https://mlr3book.mlr-org.com/train-predict.html) of the mlr3book, 80% of the penguin data set will be used for the training set and 20% for the test set:
+  
+```{r pred, message=FALSE, warning=FALSE}
+train_set = sample(task_peng$nrow, 0.8 * task_peng$nrow)
+test_set = setdiff(seq_len(task_peng$nrow), train_set)
+learner$train(task_peng, row_ids = train_set)
+prediction = learner$predict(task_peng, row_ids = test_set)
+```
+
+And the plots below show the results for `FeatureImp` and `FeatureEffect`.
+
+`FeatureImp`:
+```{r featimppred, message=FALSE, warning=FALSE, fig.cap='FeatImp train data set',  fig.align='center'}
+model = Predictor$new(learner, data = penguins[train_set,], y = "species")
+effect = FeatureImp$new(model, loss ="ce" )
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+The results of the train set for `FeatureImp` (and `FeatureEffect`) are similar to the results above, which is expected. 
+
+```{r featimptest, message=FALSE, warning=FALSE, fig.cap='FeatImp test data set', fig.align='center'}
+model = Predictor$new(learner, data = penguins[test_set,], y = "species")
+effect = FeatureImp$new(model, loss ="ce" )
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+
+Depending on which batch of input is sampled into the train data set, the reader will either see under- or overfitting in the test set.
```

Comment:
Here I would write something like "We could either use training or test data to analyze the feature effects".

Reply:
And delete the current sentence in line 139